vcd/resource_vcd_vapp.go

```diff
@@ -439,18 +480,32 @@ func resourceVcdVAppUpdate(d *schema.ResourceData, meta interface{}) error {
 }
 
 func resourceVcdVAppRead(d *schema.ResourceData, meta interface{}) error {
+	return genericVcdVAppRead(d, meta, "resource")
+}
+
+func genericVcdVAppRead(d *schema.ResourceData, meta interface{}, origin string) error {
 	vcdClient := meta.(*VCDClient)
 
 	org, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
 	if err != nil {
 		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
 	}
+	identifier := d.Id()
 
-	vapp, err := vdc.GetVAppByNameOrId(d.Id(), false)
+	if identifier == "" {
+		identifier = d.Get("name").(string)
```

Comment:
What is the purpose of this double-edged identifier? Why don't we know in the code here whether it is ID or name?

Reply:
I have used this method in other PRs. This allows us to use the same read function from both the resource and data source